hubV2/storage/provider/SPECS.MD

```diff
@@ -0,0 +1,26 @@
+#Storage Providers
```

Comment:
i don't think we should have `hub/storage/provider` or `hub/compute/provider` be the access here. these providers should be under 

- `hub/core/storage/`
  - `provider.py` has base storageprovider class
  - `s3.py` stores s3 provider
  - `local.py` stores local provider
  - `memory.py` stores memory provider
  - ...

- `hub/core/compute/`
  - `provider.py` has base compute provider class
  - `ray.py`
  - `dask.py` 
  - ...

Reply:
I would recommend moving the providers outside our core folder.

Reply:
Also "hub" folder is top level regardless. The problem or re-arranging folders inside hub won't change the problem you have mentioned. But a solution to that would be to reduce number of files we se at the top level.